base/error.jl

```diff
@@ -85,6 +85,31 @@ function _reformat_bt(bt, bt2)
     ret
 end
 
+# Same as above, but inspects a single, stacked array to parse out interepreter frames
+function _reformat_bt(bt)
+    get_ip(frame::Ptr) = UInt(frame)
+    get_ip(frame::UInt) = frame
+    get_ptr(frame::UInt) = Ptr{Cvoid}(frame)
+    get_ptr(frame::Ptr) = frame
+
+    ret = Vector{Union{InterpreterIP,Ptr{Cvoid},UInt}}()
+    i = 1
+    while i <= length(bt)
+        # Find all interpreter frames
+        if get_ip(bt[i]) == (-1)%UInt
+            push!(ret, InterpreterIP(
+                unsafe_pointer_to_objref(get_ptr(bt[i+1])),
+                bt[i+2])
+            )
```

Comment:
See also #33277 which clarifies and generalizes the bt buffer format. Also note that Jameson disabled the recording of `InterpreterIP`s in the profile traces in #33190, IIRC. (Just pointing it out in case it's relevant; I haven't read the rest of this PR in detail yet.)

Actually I think it would make sense to remove the other version of `_reformat_bt` and mostly ignore `bt2` (we would still need `bt2` as an array of roots in some cases, but that's it). I'd be happy to do that cleanup after #33277 is merged.

More generally, I wonder whether #33277 can help you with this PR. It does allow storing additional information in backtrace entries if required.

Reply:
Thanks, I'll have a look. I focused on getting the PR to build again first, I've yet to take a look at all of the recent Profile/stacktrace changes :slightly_smiling_face:

Reply:
I guess I don't need `_reformat_bt` at all if I just make sure `InterpreterIP`s aren't recorded like the profiler now does? That's what I did now, adding some more arguments to `rec_backtrace` and friends.

An extended backtrace format could be useful to store the allocation info (or a pointer to it), keeping track of the allocated type, its size, the resulting object, a tag, etc. There's a separate buffer for that now.

Reply:
Correct; if you don't want to deal with interpreter frames you can just disable the collection of those. Though it could be confusing if frames which happen to be interpreted are invisible in the profile.

Regarding storing other data in the backtrace — you can certainly do it with the "extended backtrace entries" PR. This is somewhat attractive and may simplify things if each event naturally goes with a backtrace.

Reply:
> adding some more arguments to `rec_backtrace` and friends.

I noticed that you've added an extra `incomplete` argument to those functions. That makes sense but I think it would be nicer to be consistent between those and `jl_unw_stepn` which uses the return code to signal `incomplete`. (Ie, swapping the return code with the `bt_size` argument, one way or another.)

Reply:
Good idea!